pilot/pkg/networking/util/util.go

```diff
@@ -171,9 +171,20 @@ func AddrStrToCidrRange(addr string) (*core.CidrRange, error) {
 	}, nil
 }
 
+// OptionalBuildOptions is a struct to hold optional parameters for BuildAddress and BuildAdditionalAddresses.
+type OptionalBuildAddressOptions struct {
```

Comment:
cannot you pass the transport directly, and i also do not see the need to use `...OptionalBuildAddressOptions`

Reply:
Are you proposing to update the tinker with the value after the address has been created?
E.g: 
```
for every extraAddress:
     extraAddress.Address.GetSocketAddress().Protocol = transport.ToEnvoyProtocol()
```